streaming_jupyter_integrations/magics.py

```diff
@@ -146,13 +147,28 @@ def _flink_connect_remote(self, hostname: str, port: int) -> None:
         global_conf = read_flink_config_file()
         conf = self.__create_configuration_from_dict(global_conf)
         gateway = get_gateway()
-        empty_string_array = gateway.new_array(gateway.jvm.String, 0)
+        pipeline_jars_list = self._get_pipeline_jars(conf)
         self.s_env = StreamExecutionEnvironment(
             gateway.jvm.org.apache.flink.streaming.api.environment.StreamExecutionEnvironment.createRemoteEnvironment(
-                hostname, port, conf._j_configuration, empty_string_array
+                hostname, port, conf._j_configuration, pipeline_jars_list
             )
         )
 
+    @staticmethod
+    def _get_pipeline_jars(conf: Configuration) -> JavaArray:
+        # When calling StreamExecutionEnvironment.createRemoteEnvironment() "pipeline.jars" property is overwritten
+        # with paths of jars specified as "String jarFiles" parameter. In consequence, we need to parse the property
+        # and pass the paths as "jarFiles" parameter.
+        # What is more, jarFiles are resolved incorrectly if the paths contain "file://" schema
+        # (org.apache.flink.api.java.RemoteEnvironmentConfigUtils#getJarFiles). On the other hand, jars specified in
+        # "pipeline.jars" have to contain the schema. Therefore, we need to remove the schema on our own.
+        gateway = get_gateway()
+        pipeline_jars = conf.get_string("pipeline.jars", "").split(";")
+        empty_string_array = gateway.new_array(gateway.jvm.String, len(pipeline_jars))
+        for i in range(len(pipeline_jars)):
+            empty_string_array[i] = pipeline_jars[i].replace("file://", "")
+        return empty_string_array
```

Comment:
Lets use other name for this variable since now it is no longer an empty array. Perhaps it would be good to add a unit test / update current unit tests to reflect this change?

Reply:
right, thanks!